### PR TITLE
[nrf noup] wpa_supplicant: Remove non-output status

### PR DESCRIPTION
**Pull request by kapbh**
There is "OK"/"FAIL" status multipal times during
wifi connect. Skip for wifi management commands.

### CODE REVIEW

File: wpa_supplicant/wpa_cli_zephyr.c [hunk 1]
```diff
@@ -73,16 +74,17 @@ static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, const char *cmd, int print,
 		os_memcpy(resp, buf, len - 1);
 	}
 
-	if (print) {
+	if (print_ok) {
 			buf[len] = '\0';
 			printf("%s", buf);
+			print_ok = true;
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_zephyr.c:80]: How would this work? As the initial value of `print_ok` is false it would never enter this branch, no?

File: wpa_supplicant/wpa_cli_zephyr.c [hunk 2]
```diff
@@ -35,13 +35,14 @@ struct wpa_ctrl *mon_conn;
 struct wpa_ctrl *global_ctrl_conn;
 char *ifname_prefix = NULL;
 extern struct wpa_global *global;
+bool print_ok = false;
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_zephyr.c:38]: please use `static` and then you can remove the initialisation as well.

File: wpa_supplicant/wpa_cli_zephyr.c (hunk 1 above)
Comment (kapbh) [reply wpa_supplicant/wpa_cli_zephyr.c:80]: Yes, right ! was missed out.

File: wpa_supplicant/wpa_cli_zephyr.c [hunk 3]
```diff
@@ -74,17 +75,19 @@ static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, const char *cmd, int print,
 		resp[len - 1] = '\0';
 	}
 
-	if (print) {
+	if (!print_ok) {
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_zephyr.c:78]: This would print `OK` only once, but what we want is:

* `wpa_cli` should still output OK/FAIL for each command (standard behaviour as this is interactive)
* Everything else i.e., `wifi` commands should never output OK/FAIL, not even once.

**Review by krish2718 — CHANGES_REQUESTED** [review]

**Comment by sachinthegreen** [comment]
Please change this PR to a draft till the final solution is implemented.

File: wpa_supplicant/wpa_cli_cmds.c [hunk 4]
```diff
@@ -260,7 +274,7 @@ static int wpa_cli_cmd_disable_network(struct wpa_ctrl *ctrl, int argc,
 static int wpa_cli_cmd_add_network(struct wpa_ctrl *ctrl, int argc,
 				   char *argv[])
 {
-	return wpa_ctrl_command(ctrl, "ADD_NETWORK");
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_cmds.c:263]: Can you do this in a separate commit as a pre-cursor?

File: wpa_supplicant/wpa_cli_zephyr.c [hunk 5]
```diff
@@ -78,7 +79,12 @@ static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, const char *cmd, int print,
 			buf[len] = '\0';
 			if (buf[0] != '\0')
 				printf("%s", buf);
-	}
+	}else if(!print_ok){
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_zephyr.c:82]:
```suggestion
	}else if (!print_ok){
```

File: wpa_supplicant/wpa_cli_zephyr.c [hunk 6]
```diff
@@ -78,7 +79,12 @@ static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, const char *cmd, int print,
...
+			buf[len] = '\0';
+			if (buf[0] != '\0')
+				printf("%s", buf);
+			print_ok = true;
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_zephyr.c:86]: Is this needed now?

File: wpa_supplicant/wpa_cli_cmds.c [hunk 7]
```diff
@@ -2632,7 +2646,7 @@ static int wpa_cli_cmd_twt_teardown(struct wpa_ctrl *ctrl, int argc,
 
 static int wpa_cli_cmd_erp_flush(struct wpa_ctrl *ctrl, int argc, char *argv[])
 {
-	return wpa_ctrl_command(ctrl, "ERP_FLUSH");
+	return wpa_ctrl_command_interactive(ctrl, "ERP_FLUSH");
```
Comment (krish2718) [inline wpa_supplicant/wpa_cli_cmds.c:2649]: Well, shouldn't these be `wpa_cli_cmd`?

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: wpa_supplicant/wpa_cli_cmds.c (hunk 4 above)
Comment (kapbh) [reply wpa_supplicant/wpa_cli_cmds.c:263]: Ok sure.

File: wpa_supplicant/wpa_cli_zephyr.c (hunk 6 above)
Comment (kapbh) [reply wpa_supplicant/wpa_cli_zephyr.c:86]: This change will print only one time "OK" after flashing the board. Otherwise there is blank line.

**Comment by jukkar** [comment]
Do we really need this? I mean that the OK/FAIL will show the status of the issued command which is kind of useful information.
Annoyingly the FAIL/OK print extra `\n` which looks bad, so I am proposing this change in upstream that skips the extra newline here https://github.com/zephyrproject-rtos/hostap/pull/2/commits/c3dddd36031e838fcae4a31975b96e78393db3a7

**Comment by krish2718** [comment]
> Do we really need this? I mean that the OK/FAIL will show the status of the issued command which is kind of useful information. Annoyingly the FAIL/OK print extra `\n` which looks bad, so I am proposing this change in upstream that skips the extra newline here [zephyrproject-rtos/hostap@c3dddd3](https://github.com/zephyrproject-rtos/hostap/commit/c3dddd36031e838fcae4a31975b96e78393db3a7)

Yes, we had multiple tickets from customers about the random OK/FAIL without any context. These are only useful in interactive environment where a single command is types in and you get a response OK/FAIL with newlines. But in non-interactive environment these are useless and the error checking can be handled programatically.

File: wpa_supplicant/wpa_cli_zephyr.c (hunk 6 above)
Comment (krish2718) [reply wpa_supplicant/wpa_cli_zephyr.c:86]: We can get rid of this and either print all for interactive or NONE for non-interactive.

Comment (kapbh) [reply wpa_supplicant/wpa_cli_zephyr.c:86]: Okay.

**Review by krish2718 — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]
